test/functional/testlib.bash

```diff
@@ -0,0 +1,502 @@
+	# added by default to every test environment
+	create_bundle -L -n os-core -v "$version" -f /usr/bin/core "$env_name"
+
+}
+
+# Destroys a test environment
+# Parameters:
+# - ENVIRONMENT_NAME: the name of the test environment to be deleted
+destroy_test_environment() { 
+
+	local env_name=$1
+	local deletable=true
+	validate_path "$env_name"
+	# since the action to be performed is very destructive, at least
+	# make sure the directory does look like a test environment
+	for var in "state" "target-dir" "web-dir"; do
+		if [ ! -d "$env_name/$var" ]; then
+			deletable=false
+		fi
+	done
+	if [ "$deletable" = true ]; then
+		sudo rm -rf "$env_name"
+	else
+		echo "The name provided doesn't seem to be a valid test environment"
+		return 1
+	fi
+
+}
+
+# Creates a bundle in the test environment. The bundle can contain files, directories or symlinks.
+create_bundle() { 
+
+	cb_usage() { 
+		echo "Usage:"
+		echo -e "\\tcreate_bundle [-L] [-n] <bundle_name> [-v] <version> [-d] <list of dirs> [-f] <list of files> [-l] <list of links> ENV_NAME"
+		echo "Options:"
+		echo -e "\\t-L\\tWhen the flag is selected the bundle will be 'installed' in the target-dir, otherwise it will only be created in web-dir"
+		echo -e "\\t-n\\tThe name of the bundle to be created, if not specified a name will be autogenerated"
+		echo -e "\\t-v\\tThe version for the bundle, if non selected version 10 will be used"
+		echo -e "\\t-d\\tComma-separated list of directories to be included in the bundle"
+		echo -e "\\t-f\\tComma-separated list of files to be created and included in the bundle"
+		echo -e "\\t-l\\tComma-separated list of symlinks to be created and included in the bundle,"
+		echo "Notes:"
+		echo -e "\\t- if no option is selected, a minimal bundle will be created with only one directory"
+		echo -e "\\t- for every symlink created a related file will be created and added to the bundle as well"
+		echo -e "\\t- if the '-f' or '-l' options are used, and the directories where the files live don't exist,"
+		echo -e "\\t  they will be automatically created and added to the bundle for each file"
+		echo -e "\\nExample of usage:\\n"
+		echo -e "\\tThe following command will create a bundle named 'test-bundle', which will include three directories,"
+		echo -e "\\tfour files, and one symlink (they will be added to the bundle's manifest), all these resources will also"
+		echo -e "\\tbe tarred. The bundle's manifest will be added to the MoM."
+		echo -e "\\n\\tcreate_bundle -n test-bundle -f /usr/bin/test-1,/usr/bin/test-2,/etc/systemd/test-3 -l /etc/test-link my_test_env"
+		echo -e "\\n"
+		
+	}
+	
+	local OPTIND
+	local opt
+	local dir_list
+	local file_list
+	local link_list
+	local version
+	local bundle_name
+	local env_name
+	local files_path
+	local version_path
+	local manifest
+	local local_bundle=false
+
+	set -f  # turn off globbing
+	while getopts :v:d:f:l:n:L opt; do
+		case "$opt" in
+			d)	IFS=,
```

Comment:
I really don't like changing IFS globally, and even less so not either unsetting it or setting it back to a sane value after you have parsed the arguments. A helper function to read an array would be good, and in that you could declare IFS to be local.

Reply:
what about just defining IFS local for this function?

Reply:
Defining for just this junction would be better, but at about 170 lines long for this function, and it calling other functions, I still feel that the IFS change has too much scope.